### ISSUE TITLE
illegal move made

### ISSUE THREAD

**Issue by Trey2k** [WeaselChess/Weasel#18]
```
2021-02-13 20:39:53.285-->1:position startpos moves d2d4 g8f6 c2c4 e7e6 g1f3 d7d5 e2e3 f8e7 b1c3 e8g8 d1c2 c7c5 c4d5 c5d4 e3d4 e6d5 f1e2 b8c6 e1g1 f8e8 c1e3 c8g4 f1e1 a8c8 f3e5 g4e2 e1e2 f6e4 e5c6 c8c6 c2b3 e4c3 b2c3 c6b6 b3a4 b6a6 a4b3 a6b6
2021-02-13 20:39:53.285-->1:go wtime 11061 btime 18674 winc 1000 binc 1000
2021-02-13 20:39:53.287<--1:11061
2021-02-13 20:39:53.289<--1:time: 368 start: 1613266793285 stop: 1613266794653 depth 64 timeset true
2021-02-13 20:39:53.291<--1:info score cp -5 depth 1 nodes 106 time 1 pv b3c2
2021-02-13 20:39:53.293<--1:info score cp -5 depth 2 nodes 431 time 1 pv b3a4 b6a6
2021-02-13 20:39:53.295<--1:info score cp -2 depth 3 nodes 1037 time 1 pv b3a4 b6a6 a4b3
2021-02-13 20:39:53.297<--1:info score cp -2 depth 4 nodes 2749 time 2 pv b3a4 b6a6 a4b3
2021-02-13 20:39:53.300<--1:info score cp -2 depth 5 nodes 6732 time 4 pv b3a4 b6a6 a4b3
2021-02-13 20:39:53.303<--1:info score cp -2 depth 6 nodes 16004 time 8 pv b3a4 b6a6 a4b3
2021-02-13 20:39:53.332<--1:info score cp -2 depth 7 nodes 103393 time 45 pv b3a4 b6a6 a4b3
2021-02-13 20:39:53.400<--1:info score cp -2 depth 8 nodes 276512 time 113 pv b3a4 b6a6 a4b3
2021-02-13 20:39:54.041<--1:info score cp -2 depth 9 nodes 1815930 time 754 pv b3a4 b6a6
2021-02-13 20:39:54.655<--1:bestmove b3a4
2021-02-13 20:39:54.655*1*Found move:Qb3-a4
2021-02-13 20:39:54.815*2*Child Process Prio Adj:  PID 25200 conhost.exe
2021-02-13 20:39:54.816*2*Start calc, move no: 39
2021-02-13 20:39:54.871-->2:time 1868
2021-02-13 20:39:54.871-->2:otim 1070
2021-02-13 20:39:54.871-->2:b3a4
2021-02-13 20:39:54.873<--2:1 0 0 95 d8b8
2021-02-13 20:39:54.875<--2:2 0 0 414 b6a6
2021-02-13 20:39:54.877<--2:3 0 0 2650 b6a6
2021-02-13 20:39:54.888<--2:4 0 1 19263 b6a6
2021-02-13 20:39:54.979<--2:5 0 10 146344 b6a6
2021-02-13 20:39:55.495<--2:move b6a6
2021-02-13 20:39:55.495*2*Found move:Rb6-a6
2021-02-13 20:39:55.634*1*Child Process Prio Adj:  PID 25544 conhost.exe
2021-02-13 20:39:55.636*1*Start calc, move no: 40
2021-02-13 20:39:55.637-->1:position startpos moves d2d4 g8f6 c2c4 e7e6 g1f3 d7d5 e2e3 f8e7 b1c3 e8g8 d1c2 c7c5 c4d5 c5d4 e3d4 e6d5 f1e2 b8c6 e1g1 f8e8 c1e3 c8g4 f1e1 a8c8 f3e5 g4e2 e1e2 f6e4 e5c6 c8c6 c2b3 e4c3 b2c3 c6b6 b3a4 b6a6 a4b3 a6b6 b3a4 b6a6
2021-02-13 20:39:55.637-->1:go wtime 10691 btime 19070 winc 1000 binc 1000
2021-02-13 20:39:55.639<--1:10691
2021-02-13 20:39:55.641<--1:time: 356 start: 1613266795638 stop: 1613266796994 depth 64 timeset true
2021-02-13 20:39:55.643<--1:info score cp -2 depth 1 nodes 78 time 0 pv a4b3
2021-02-13 20:39:55.645<--1:info score cp -2 depth 2 nodes 315 time 0 pv a4b3 d8d7
2021-02-13 20:39:55.647<--1:info score cp 0 depth 3 nodes 700 time 0 pv a4b3
2021-02-13 20:39:55.650<--1:info score cp -2 depth 4 nodes 2604 time 1 pv a4b3 d8d7 a1e1 a6e6
2021-02-13 20:39:55.653<--1:info score cp 0 depth 5 nodes 9211 time 4 pv a4b3
2021-02-13 20:39:55.686<--1:info score cp 0 depth 6 nodes 118856 time 46 pv a4b5 a6b6
2021-02-13 20:39:55.757<--1:info score cp 0 depth 7 nodes 277140 time 118 pv a4b5 a6b6
2021-02-13 20:39:55.931<--1:info score cp 0 depth 8 nodes 660084 time 292 pv a4b5 a6b6
2021-02-13 20:39:56.573<--1:info score cp 0 depth 9 nodes 2107523 time 934 pv a4b5 a6b6
2021-02-13 20:39:56.996<--1:bestmove a4b5
2021-02-13 20:39:56.996*1*Found move:Qa4-b5
2021-02-13 20:39:57.153*2*Child Process Prio Adj:  PID 25200 conhost.exe
2021-02-13 20:39:57.154*2*Start calc, move no: 41
2021-02-13 20:39:57.215-->2:time 1907
2021-02-13 20:39:57.215-->2:otim 1034
2021-02-13 20:39:57.215-->2:a4b5
2021-02-13 20:39:57.217<--2:1 0 0 316 a6b6
2021-02-13 20:39:57.220<--2:2 0 0 730 a6b6 b5a5
2021-02-13 20:39:57.222<--2:3 0 0 4111 a6b6 b5a5 b6b1
2021-02-13 20:39:57.236<--2:4 0 2 25607 a6b6 b5a4
2021-02-13 20:39:57.331<--2:5 0 11 156173 a6b6 b5a4
2021-02-13 20:39:57.852<--2:move a6b6
2021-02-13 20:39:57.852*2*Found move:Ra6-b6
2021-02-13 20:39:58.005*1*Child Process Prio Adj:  PID 25544 conhost.exe
2021-02-13 20:39:58.006*1*Start calc, move no: 42
2021-02-13 20:39:58.007-->1:position startpos moves d2d4 g8f6 c2c4 e7e6 g1f3 d7d5 e2e3 f8e7 b1c3 e8g8 d1c2 c7c5 c4d5 c5d4 e3d4 e6d5 f1e2 b8c6 e1g1 f8e8 c1e3 c8g4 f1e1 a8c8 f3e5 g4e2 e1e2 f6e4 e5c6 c8c6 c2b3 e4c3 b2c3 c6b6 b3a4 b6a6 a4b3 a6b6 b3a4 b6a6 a4b5 a6b6
2021-02-13 20:39:58.007-->1:go wtime 10336 btime 19450 winc 1000 binc 1000
2021-02-13 20:39:58.009<--1:10336
2021-02-13 20:39:58.011<--1:time: 344 start: 1613266798007 stop: 1613266799351 depth 64 timeset true
2021-02-13 20:39:58.013<--1:info score cp 0 depth 1 nodes 85 time 0 pv
2021-02-13 20:39:58.015<--1:info score cp 0 depth 2 nodes 292 time 0 pv
2021-02-13 20:39:58.018<--1:info score cp 0 depth 3 nodes 436 time 0 pv
2021-02-13 20:39:58.021<--1:info score cp 0 depth 4 nodes 1193 time 1 pv
2021-02-13 20:39:58.024<--1:info score cp 0 depth 5 nodes 7837 time 4 pv
2021-02-13 20:39:58.027<--1:info score cp 0 depth 6 nodes 18278 time 7 pv
2021-02-13 20:39:58.030<--1:info score cp 0 depth 7 nodes 46426 time 21 pv
2021-02-13 20:39:58.059<--1:info score cp 0 depth 8 nodes 132697 time 51 pv
2021-02-13 20:39:58.854<--1:info score cp 0 depth 9 nodes 2026609 time 846 pv
2021-02-13 20:39:59.354<--1:bestmove a4b5
2021-02-13 20:39:59.354*1*---------> Arena:Illegal move!: "a4b5" (Feinprüfung)
2021-02-13 20:39:59.361****Tour**Game end: 0-1 {0-1 Arena Adjudication. Illegal move!}
2021-02-13 20:39:59.361****Tour**Last game won by Tscp181 with Black in 21 moves
2021-02-13 20:39:59.362****Tour**Weasel  -  Tscp181: 
2021-02-13 20:39:59.362****Tour**  19 B 2    2.0:1.0        16.0:3.0      8:39:59 PM     0.00    21 7.909 :5.000  15-2-2
2021-02-13 20:39:59.426****Tour**Saving file C:\Chess\Arena\Tournaments\Weasel Test 8.at.pgn Round:19 2021.02.13 20:39:28
2021-02-13 20:39:59.503****Tour**stopping engine calculation after game...
2021-02-13 20:39:59.504-->2:?
2021-02-13 20:39:59.504<--2:Error (unknown command): ?
2021-02-13 20:39:59.527-->2:result 0-1
2021-02-13 20:39:59.527<--2:Error (unknown command): result
2021-02-13 20:40:00.673****Tour**
2021-02-13 20:40:00.673****Tour**Pairing 1/7: Weasel  -  Horizon_4_4
```

**Comment by Trey2k** [comment]
```
position startpos moves d2d4 g8f6 c2c4 e7e6 g1f3 d7d5 e2e3 f8e7 b1c3 e8g8 d1c2 c7c5 c4d5 c5d4 e3d4 e6d5 f1e2 b8c6 e1g1 f8e8 c1e3 c8g4 f1e1 a8c8 f3e5 g4e2 e1e2 f6e4 e5c6 c8c6 c2b3 e4c3 b2c3 c6b6 b3a4 b6a6 a4b3 a6b6 b3a4 b6a6 a4b5 a6b6
print

Board State:

8  .  .  .  q  r  .  k  .
7  p  p  .  .  b  p  p  p
6  .  r  .  .  .  .  .  .
5  .  Q  .  p  .  .  .  .
4  .  .  .  P  .  .  .  .
3  .  .  P  .  B  .  .  .
2  P  .  .  .  R  P  P  P
1  R  .  .  .  .  .  K  .
   A  B  C  D  E  F  G  H
Side: w
EnPassant: None
Castel Perms: ----
Position Hash: B82EA396D8578EFA
Is Repition: false
Evaluation CP: -2
Fifty Move Count: 9
HispPly: 43
go wtime 10336 btime 19450 winc 1000 binc 1000
10336
time: 344 start: 1613267064027 stop: 1613267065371 depth 64 timeset true
info score cp 0 depth 1 nodes 85 time 0 pv
info score cp 0 depth 2 nodes 409 time 1 pv
info score cp 0 depth 3 nodes 2285 time 2 pv
info score cp 0 depth 4 nodes 8404 time 12 pv
info score cp 0 depth 5 nodes 31101 time 31 pv
info score cp 0 depth 6 nodes 101218 time 72 pv
info score cp 0 depth 7 nodes 243171 time 145 pv
info score cp 0 depth 8 nodes 879565 time 401 pv
info score cp 0 depth 9 nodes 2103018 time 943 pv
panic: runtime error: index out of range [100] with length 13

goroutine 38 [running]:
github.com/WeaselChess/Weasel/engine/board.(*PositionStruct).hashPiece(...)
        C:/Development/Go/src/github.com/WeaselChess/Weasel/engine/board/hashkeys.go:68
github.com/WeaselChess/Weasel/engine/board.(*PositionStruct).movePiece(0xa2e3a0, 0x0, 0x0, 0x0, 0x0)
        C:/Development/Go/src/github.com/WeaselChess/Weasel/engine/board/movement.go:181 +0x446
github.com/WeaselChess/Weasel/engine/board.(*PositionStruct).MakeMove(0xa2e3a0, 0x0, 0x7530, 0xa, 0x1)
        C:/Development/Go/src/github.com/WeaselChess/Weasel/engine/board/makemove.go:118 +0x32c
github.com/WeaselChess/Weasel/engine/search.(*InfoStruct).SearchPosition(0xa5f520, 0xa2e3a0, 0x9531b1, 0x30)
        C:/Development/Go/src/github.com/WeaselChess/Weasel/engine/search/search.go:400 +0x7e5
github.com/WeaselChess/Weasel/engine/uci.goHandler(0xc0000b6000, 0x9, 0x9)
        C:/Development/Go/src/github.com/WeaselChess/Weasel/engine/uci/handlers.go:165 +0xe87
created by github.com/WeaselChess/Weasel/engine/uci.UCI
        C:/Development/Go/src/github.com/WeaselChess/Weasel/engine/uci/uci.go:80 +0x325
exit status 2
``` 
Some more debug info, ill tackle this tomrow

**Comment by Trey2k** [comment]
Fixed in 177d94bfa0274d1550553146a3b6f0075ca26474
